### PR TITLE
[IMP] hr_recruitment(_skills/_survey): grant interviewer more rights

### PR DESCRIPTION
**Pull request by sofiagvaladze**
Purpose: In order to increase visibility of the interviewer group, it is moved from hidden group, to recruitment group.

Moreover, we grant the interviewer additional rights in order to manage the application process properly and be able to execute following actions • Send an interview
• Check interview results
• Add/delete/modify skills
• Create a meeting

In order to allow above mentioned actions on skills to the interviewer, we grant them all the rights to the hr_applicant_skill.

In order to let the interviewer send the survey we grant them access to the private addresses and now recruitment-interviewer implies survey-officer group as well.

Additionally, for the security purposes, we make a change, so that the survey officer is not able to read recruitment interview results unless they are also an interviewer for a specific applicant.

task - 3179332

### CODE REVIEW

File: addons/hr_recruitment_survey/security/hr_recruitment_survey_security.xml [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+        <record id="hr_recruitment.group_hr_recruitment_interviewer" model="res.groups">
+            <field name="implied_ids" eval="[Command.link(ref('survey.group_survey_user')), Command.link(ref('base.group_private_addresses'))]"/>
+            <field name="comment">The recruiter interviewer will be able to send survey to the applicant.</field>
+        </record>
+        <record id="survey.survey_user_input_rule_survey_user_read" model="ir.rule">
+            <field name="name">Survey user input: officer: read all non private survey answers</field>
+            <field name="domain_force">[('applicant_id', '=', False)]</field>
+        </record>
+        <record id="survey_user_input_rule_recruitment_interviewer" model="ir.rule">
+            <field name="name">Survey user input: recruitment interview: read survey answers that they are set as an interviewer</field>
```
Comment (Bertrand2) [inline addons/hr_recruitment_survey/security/hr_recruitment_survey_security.xml:13]:
```suggestion
            <field name="name">Survey user input: recruitment interview: read survey answers for which they are set as interviewer</field>
```

**Review by Bertrand2 — APPROVED** [review]
LGTM! Apart from the comments discussed on the task, there's only some nitpicking on the english :ok_hand:

File: addons/hr_recruitment_skills/security/hr_recruitment_skills_security.xml [hunk 2]
```diff
@@ -11,4 +11,12 @@
         ]</field>
         <field name="groups" eval="[(4, ref('hr_recruitment.group_hr_recruitment_interviewer'))]"/>
     </record>
+
+    <record id="hr_applicant_skill_interviewer_rule" model="ir.rule">
+        <field name="name">Applicant Skill: Interviewer</field>
```
Comment (kba-odoo) [inline addons/hr_recruitment_skills/security/hr_recruitment_skills_security.xml:16]:
```suggestion
        <field name="name">Applicant Skill: Officer</field>
```

File: addons/hr_recruitment_survey/security/hr_recruitment_survey_security.xml [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+            <field name="comment">It will also allow to send surveys and see the resume.</field>
+        </record>
+        <record id="survey.survey_user_input_rule_survey_user_read" model="ir.rule">
+            <field name="name">Survey user input: officer: read all non private survey answers</field>
+            <field name="domain_force">[('applicant_id', '=', False)]</field>
+        </record>
+        <record id="survey_user_input_rule_recruitment_interviewer" model="ir.rule">
+            <field name="name">Survey user input: recruitment interview: read survey answers for which they are set as interviewer</field>
+            <field name="model_id" ref="survey.model_survey_user_input"/>
+            <field name="domain_force">[('applicant_id', '!=', False), ('applicant_id.interviewer_ids', 'in', user.id)]</field>
```
Comment (kba-odoo) [inline addons/hr_recruitment_survey/security/hr_recruitment_survey_security.xml:15]: Shouldn't you add `applicant_id.job_id.interviewer_ids` as well?

**Review by kba-odoo — CHANGES_REQUESTED** [review]

**Comment by kba-odoo** [comment]
robodoo r+

**Review by lm4649 — COMMENTED** [review]
Hi @tde-banana-odoo, @sofiagvaladze 
Thank you for your reviews! :smile: :prey
I've updated the code according to your suggestions. 
Concerning the fact that the interviewers own the survey users rights, I don't have a third way to propose. Does leaving things as they are pose potential security problems?

**Pull request closed without merging**